cfme/services/myservice/ui.py

```diff
@@ -147,115 +148,125 @@ def is_displayed(self):
+    wait_for(
+        lambda: view.lifecycle_detail.get_text_of("Retirement State") == 'Retiring',
+        fail_func=tb.refresh,
+        num_sec=5 * 60,
+        delay=5,
+        message='Service Retirement'
+    )
+
+
+@MyService.update.external_implementation_for(ViaUI)
+def update(self, updates):
+    view = navigate_to(self, 'Edit')
+    changed = view.fill_with(updates, on_change=view.save_button, no_change=view.cancel_button)
+    view.flash.assert_no_error()
+    if changed:
+        view.flash.assert_success_message('Service "{}" was saved'.format
+            (updates.get('name', self.name)))
+    else:
+        view.flash.assert_success_message('Edit of Service "{}" was cancelled by the \
+            user'.format(updates.get('description', self.description)))
+    view = self.create_view(MyServiceDetailView, override=updates)
+    assert view.is_displayed
+
+
+@MyService.exists.external_implementation_for(ViaUI)
+def exists(self):
+    try:
+        navigate_to(self, 'Details')
+        return True
+    except CandidateNotFound:
+        return False
+
+
+@MyService.delete.external_implementation_for(ViaUI)
+def delete(self):
+    view = navigate_to(self, 'Details')
+    view.configuration.item_select(
+        version.pick({
+            version.LOWEST: 'Remove Service from the VMDB',
+            '5.7': 'Remove Service'}),
+        handle_alert=True)
+    view = self.create_view(MyServicesView)
+    view.flash.assert_no_error()
+    assert view.is_displayed
+    view.flash.assert_success_message(
+        'Service "{}": Delete successful'.format(self.name))
+
+
+@MyService.set_ownership.external_implementation_for(ViaUI)
+def set_ownership(self, owner, group):
+    view = navigate_to(self, 'SetOwnership')
+    view.fill({'select_owner': owner,
+               'select_group': group})
+    view.save_button.click()
+    view = self.create_view(MyServiceDetailView)
+    assert view.is_displayed
+    view.flash.assert_no_error()
+    view.flash.assert_success_message('Ownership saved for selected Service')
+
+
+@MyService.edit_tags.external_implementation_for(ViaUI)
+def edit_tags(self, tag, value):
+    view = navigate_to(self, 'EditTags')
+    view.fill({'select_tag': tag,
+               'select_value': value})
+    view.save_button.click()
+    view = self.create_view(MyServiceDetailView)
+    assert view.is_displayed
+    view.flash.assert_no_error()
+    view.flash.assert_success_message('Tag edits were successfully saved')
+
+
+@MyService.check_vm_add.external_implementation_for(ViaUI)
+def check_vm_add(self, add_vm_name):
+    view = navigate_to(self, 'Details')
+    # TODO - replace Quadicon later
+    quadicon = Quadicon(add_vm_name, "vm")
+    sel.click(quadicon)
+    view.flash.assert_no_error()
+
+
+@MyService.download_file.external_implementation_for(ViaUI)
+def download_file(self, extension):
+    view = navigate_to(self, 'All')
+    view.download_choice.item_select("Download as " + extension)
```

Comment:
Prefer ``format`` for string concatenation.

Reply:
Done

Reply:
Still using ``+``.